src/FSharp.AWS.DynamoDB/Utils/Utils.fs

```diff
@@ -218,29 +216,29 @@ module internal Utils =
             else None
         | _ -> None
 
-    type Task with
-        /// Gets the inner exception of the faulted task.
-        member t.InnerException =
-            let e = t.Exception
-            if e.InnerExceptions.Count = 1 then e.InnerExceptions.[0]
-            else
-                e :> exn
-
     type Async with
+
         /// Raise an exception
         static member Raise e = Async.FromContinuations(fun (_,ec,_) -> ec e)
 
+        (* Direct copies of canonical implementation at http://www.fssnip.net/7Rc/title/AsyncAwaitTaskCorrect
+           pending that being officially packaged somewhere or integrated into FSharp.Core https://github.com/fsharp/fslang-suggestions/issues/840 *)
+
         /// <summary>
         ///     Gets the result of given task so that in the event of exception
         ///     the actual user exception is raised as opposed to being wrapped
         ///     in a System.AggregateException.
         /// </summary>
         /// <param name="task">Task to be awaited.</param>
+        [<System.Diagnostics.DebuggerStepThrough>]
         static member AwaitTaskCorrect(task : Task<'T>) : Async<'T> =
-            Async.FromContinuations(fun (sc,ec,cc) ->
-                task.ContinueWith(fun (t : Task<'T>) -> 
-                    if task.IsFaulted then ec t.InnerException 
```

Comment:
when I glanced at this in analyzing what became https://github.com/dotnet/fsharp/issues/13165, I went down a rathole of what would happen if this threw and/or returned `null`, and what that might lead to within `Async.FromContinuations`. For this reason I feel it's best to have this inlined and straightforward (along with the other benefits of it being identical to the canonical version)